litestar/datastructures/secret_values.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+
+class SecretValue:
```

Comment:
I also think this could be better if it was generic on the type of the secret value:

```py
_SecretT = TypeVar("_SecretT", bound="str | bytes")


class SecretValue(Generic[_SecretT]):
    def get_value(self) -> _SecretT: ...
    ...

class SecretString(SecretValue[str]): ...
```
This way, calling `get_value()` and other methods on a sub class won't by typed as returning the union `str | bytes`, instead it will be narrowed to the type that the subtype has been narrowed with.